### ISSUE TITLE
[BUG] Getting incomplete video title

### ISSUE THREAD

**Issue by danielhuici** [pytube/pytube#1783]
**Describe the bug**

When trying to get a video title from some of the provided URL's using `title` attribute, the string obtained does not match with the title seen on YouTube.

**To Reproduce**
Use PyTube to get the title of the following videos:

```
from pytube YouTube

print(YouTube("https://www.youtube.com/watch?v=CyT-DaI0rqQ").title) 
print(YouTube("https://www.youtube.com/watch?v=JIoj1RYvz1Y").title) 
print(YouTube("https://www.youtube.com/watch?v=YntDxln7w5U").title) 
```

**Expected behavior**
The following strings should be printed:
```
Marwa Loud - Bad Boy (Clip Officiel)
Rakky Ripper - TRACCIÓN (Official Lyric Video) - Benidorm Fest 2023
Timmy Trumpet x INNA x Love Harder - Blow It Up (Official Music Video)
```

**Output**
But the following strings are printed:
```
Bad boy
TRACCIÓN
Blow It Up
```
Looks like this is happening on some music videos, still can't guess the reason.

**System information**
Please provide the following information:
- Python version 3.10.12
- Pytube version 15.0.0
- Command used to install pytube: `pip3 install pytube`

**Comment by Zebra-0** [comment]
I had the same issue, try this:
from urllib.request import urlopen

def title(url):
    page = urlopen(url)
    html_bytes = page.read()
    html = html_bytes.decode("utf-8")

    title_index = html.find("<title>")
    start_index = title_index + len("<title>")
    end_index = html.find("</title>")
    title = html[start_index:end_index]
    if "- YouTube" in title:
        full_title = title.replace("- YouTube", "")

    return str(full_title)


print(title("https://youtu.be/NHffhGkpgFAA?list=RDAHukwv_VX9A"))

I hope this work for you

**Comment by danielhuici** [comment]
> I had the same issue, try this: from urllib.request import urlopen
> 
> def title(url): page = urlopen(url) html_bytes = page.read() html = html_bytes.decode("utf-8")
> 
> ```
> title_index = html.find("<title>")
> start_index = title_index + len("<title>")
> end_index = html.find("</title>")
> title = html[start_index:end_index]
> if "- YouTube" in title:
>     full_title = title.replace("- YouTube", "")
> 
> return str(full_title)
> ```
> 
> print(title("https://youtu.be/NHffhGkpgFAA?list=RDAHukwv_VX9A"))
> 
> I hope this work for you

Yes! This may do the job as a workaround. But I think pytube maintainers should check what's going on.

**Comment by Zebra-0** [comment]
acctually i think this is a good thing, not a bug...
for example, if you want to add the artist of the file, you can easily get the artist name subtracting url.title from the full title...

but i got your point.  would be a good get both just from pytube

**Comment by Zebra-0** [comment]
just found out a better way...
I look up into the main file and got to the vid_info function.
there you can get both, artist and the name of the music.
when we call .title, pytube get this information from vid_info dict, which gets information from an api in Innertube...
we can call
`Youtube(url).title
` or 
`info = Youtube(url).vid_info`
`title = info['videoDetails']['title']
`and artist name: 
`artist = info['videoDetails']['author']
`
`full_title = artist + ' - ' +  title`
I hope this works better for you

**Comment by danielhuici** [comment]
> just found out a better way... I look up into the main file and got to the vid_info function. there you can get both, artist and the name of the music. when we call .title, pytube get this information from vid_info dict, which gets information from an api in Innertube... we can call `Youtube(url).title ` or `info = Youtube(url).vid_info` `title = info['videoDetails']['title'] `and artist name: `artist = info['videoDetails']['author'] ` `full_title = artist + ' - ' + title` I hope this works better for you

Thanks! At least this may work for those mentioned videos, to get a more accurate title, but we can't apply this on all the scenarios. I see that this is happening on some of the official artist channels, but there are music videos whose channel name won't match the artist name (as there are many songs that are not being uploaded by the official channels) and in those cases, the artist name is useless.

**Comment by JuanBindez** [comment]
### go to __main__.py line 333 and delete the title() method and put this one in its place (package already with these fixes in "pip install pytubefix"):


          @property
              def title(self) -> str:
                  """Get the video title.
          
                  :rtype: str
                  """
                  self._author = self.vid_info.get("videoDetails", {}).get(
                      "author", "unknown"
                  )
          
                  if self._title:
                      return self._author + self._title
          
                  try:
                      self._title = self.vid_info['videoDetails']['title'] #+ self._author
                  except KeyError:
                      # Check_availability will raise the correct exception in most cases
                      #  if it doesn't, ask for a report.
                      self.check_availability()
                      raise exceptions.PytubeError(
                          (
                              f'Exception while accessing title of {self.watch_url}. '
                              'Please file a bug report at https://github.com/pytube/pytube'
                          )
                      )
                  # print(self.vid_info['videoDetails'])
                  return self._author + self._title
          
----------
### This will return the full name of the video title, but prefer to use the already corrected package
download with "pip install pytubefix", I'm fixing these bugs there
